src/uwtools/rocoto.py

```diff
@@ -183,12 +183,25 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
         :param config: Configuration data for this element.
         """
         e = SubElement(e, STR.dependency)
+        # block no longer relevant in this loop
+        # will be used in specific methods itself
+        for key in config.keys():
+            tag, _ = self._tag_name(key)
+            if tag in [STR.taskdep, STR.datadep, STR.timedep]:
+                self._add_task_dependency_operand(e, config=config)
+            else:
+                raise UWConfigError("Unhandled dependency type %s" % tag)
+
+    def _add_task_dependency_operand(self, e: Element, config: dict) -> None:
+        """
+        ???
+        """
         for key, block in config.items():
             tag, _ = self._tag_name(key)
             if tag == STR.taskdep:
                 self._set_attrs(SubElement(e, STR.taskdep), block)
-            else:
-                raise UWConfigError("Unhandled dependency type %s" % tag)
+            elif tag == STR.datadep:
+                pass
```

Comment:
Still trying to figure out this one. I want to set the attrs on `datadep` however the `taskdep` currently calls `_set_attrs` and within that method we see `config[STR.attrbs].items()`. In `datatdep` we will not see an `attrbs` element making it hard to navigate currently.

Reply:
Maybe you mentioned that there _are_ attributes to set, but that they're just not in a sub-dict called `attrs`. Is that right? If so, you might be able to still use `_set_attrs()` by constructing a dict literal, populated from the attributes in `block`, and passing it in via the `config` parameter. So, the call here might look like
```
self._set_attrs(SubElement(e, STR.datadep), config={"attrs": {???: block[???], ???: block[???], ...}})
```
where the `???`s are keys in the `datadep` block. Is that realistic?

Reply:
> adding an `attrs` key to the dict
Maybe be careful about actually modifying the original config dict, as that could have unintended consequences elsewhere. You could either use `deepcopy()` from Python's `copy` module to get a copy that would be safe to mutate (and then discard later), or construct a small dict on the fly containing just what you need and passing that down.